### DISCUSSION TITLE
How learnlib works

### DISCUSSION THREAD

**Discussion by VishalPatel43** [LearnLib/learnlib#96] [Q&A] [answered]
1) How to run this learnlib or can you provide the tutorial or docs where to learn
2)Is there any method which generate the random DFAs (dot file)
3) Finally I want to make a tool using L*, TTT, KV algorithms which takes the dot file of DFAs  and compare their membership query, equivalance query etc on Random DFAs
Please give any suggestions how to do all this

**Comment by mtf90** [comment, marked as answer]
1) We have an [examples](https://github.com/LearnLib/learnlib/tree/develop/examples) module in the repository which should give you ready-to-use starting points on how to work with (the different features of) LearnLib. If you have any specific questions about one of the concepts/components, feel free to ask.

2) The `automata-util` module holds a `RandomAutomata` class that allows you to generate random automata of different kinds (such as DFAs). If you want to (de-)serialize DFAs to (from) DOT, there exists an `automata-serialization-dot` module that provides a `DOTParsers` and `GraphDOT` class for (de-) serializing automata. You may have a look at the [test cases](https://github.com/LearnLib/automatalib/tree/develop/serialization/dot/src/test) to see how to operate them. Again, if you have any specific questions, just ask.


Unfortunately, some of the code is only available in the current development version. However, we are currently preparing a release which hopefully arrives in 2-3 weeks. So you could start to familiarize yourself with the code using the current development version and then use the new release in your project later.

**Reply by VishalPatel43** [reply]
SIr I am not able to find automata-util module in learnlib

**Reply by mtf90** [reply]
The `automata-` modules are part of [AutomataLib](https://github.com/LearnLib/automatalib/wiki/List-of-AutomataLib-Artifacts).
